src/gwt/src/org/rstudio/studio/client/rsconnect/ui/RSConnectDeploy.java

```diff
@@ -154,6 +162,97 @@ public enum ServerType {
       SHINYAPPS
    }
    
+   public static class EnvironmentVariablesDialog extends ModalDialog<List<String>>
```

Comment:
Minor nit based on the screenshot: when in the "No variables found" mode, be nice to not still have both OK and Cancel buttons, just one button (Ok or Close?). Not worth reworking things if there isn't a trivial solution.

Reply:
Good idea; I'll see how straightforward that is.